cmd/syncComparator.go

```diff
@@ -82,16 +83,18 @@ type syncSourceComparator struct {
 	// storing the destination objects
 	destinationIndex *objectIndexer
 
+	preferSMBTime     bool
 	disableComparison bool
 }
 
-func newSyncSourceComparator(i *objectIndexer, copyScheduler objectProcessor, disableComparison bool) *syncSourceComparator {
-	return &syncSourceComparator{destinationIndex: i, copyTransferScheduler: copyScheduler, disableComparison: disableComparison}
+func newSyncSourceComparator(i *objectIndexer, copyScheduler objectProcessor, preferSMBTime, disableComparison bool) *syncSourceComparator {
```

Comment:
we really should start creating options bags :'(

Reply:
Yes I completely agree.

I would like to take some time in the near future to focus on improving the overall quality of life in the AzCopy codebase, because I generally feel immense pain whenever I have to pass a flag through 8 layers of code and sometimes with 20+ argument functions